lint-staged.config.js

```diff
@@ -0,0 +1,4 @@
+
+module.exports = {
+    "**/*.ts": ["npx eslint --fix", "prettier --no-config --write", "tsc", "git add dist"]
```

Comment:
can we move this config into `package.json` (just like how husky is also there)

it just needs to be keyed using `"lint-staged"`

Reply:
additionally, considering eslint and prettier are being run across every file in the project, perhaps using the glob `"*"` would be more representative of running `npm run format` than `"**/*.ts"`

Reply:
can we also use `npx tsc`?

Reply:
and lastly, wouldn't `git add` be better?

Reply:
> and lastly wouldn't `git add` be better?
Well `lint-stage` automatically uses `git add` but it only restages the files that have already been staged, so we still need to stage `dist` in order for it to go in the commit.

Reply:
ah got it, that makes sense 👍